### PR TITLE
Add commit fixup and rebase with autosquash

### LINKED ISSUE #410: Commit context menu items to create fixup, squash and another for autosquash
**Describe the feature that you'd like**
I want to be able to right click on a commit and select `fixup` or `squash` to run the equivalent of running `git commit --<fixup/squash> <commit hash>`. 

I also want a context menu to run the equivalent of `git rebase -i --autosquash <commit hash>`.

**Additional context (optional)**
It might be easier, if possible, to allow creating custom context commands.

### PR DESCRIPTION
**Pull request by bkmeneguello**
Issue Number / Link:

closes #410 

Summary of the issue:

Adds an option to create a fixup commit and also adds an option to autosquash interactive rebasing.

### CODE REVIEW

**Review by lucasclerisse — APPROVED** [review]

**Comment by bkmeneguello** [comment]
Hi @mhutchie, any chance you check this? I use this feature daily basis and would be awesome to have it in vscode. Also, I have more like this to add if I could get it approved.

**Comment by Ghostbird** [comment]
@mhutchie I'd love to see this merged, do you have time to have a look at this?
@bkmeneguello I see some issues with this PR (keep in mind, I've no power here, I'm just a user who'd like this feature)

1. You have enabled _vigilant_ mode but have not signed this commit.
2. It is actually two separate features in one
    - Control over the `autosquash` flag on rebase.
    - A `fixup` commit context menu option.
3. I think the autosquash doesn't account for `git config`

Personally I don't care for the autosquash option, and I don't see it load the initial value from `git config`, nor override it. I think the actual logic executed may differ significantly from what the user selects. I don't know many use cases for _not_ autosquashing, I always enable it by default system-wide. I've performed thousands of rebases and hundreds of fixups in the last few years, and I don't remember ever having to clear the `fixup` set by default autosquash.

The fixup context menu entry really adds something significant though. It's the one command that I regularly use that I have to run from the terminal.

In conclusion, I'd suggest splitting the features of this PR across two separate ones. Because the fixup option in my opinion is more desirable, and easier to merge.

**Comment by ShuXiao2023** [comment]
This is really a great feature, and I hope this can be merge too~~
